nixos/modules/services/networking/jitsi-meet.nix

```diff
@@ -0,0 +1,354 @@
+
+          The following extra settings need to be added when running behind NAT:
+
+          org.ice4j.ice.harvest.NAT_HARVESTER_LOCAL_ADDRESS = <replaceable>Local.IP.Address</replaceable>
+
+          org.ice4j.ice.harvest.NAT_HARVESTER_PUBLIC_ADDRESS = <replaceable>Public.IP.Address</replaceable>
+        '';
+      };
+
+      openFirewall = mkOption {
+        type = bool;
+        default = false;
+        description = ''
+          Whether to open ports in the firewall for the videobridge.
+        '';
+      };
+    };
+
+    jicofo = {
+      config = mkOption {
+        type = attrsOf str;
+        default = {};
+        example = literalExample ''
+          {
+            "org.jitsi.jicofo.auth.URL" = "XMPP:jitsi-meet.example.com";
+          }
+        '';
+        description = ''
+          Contents of the <filename>sip-communicator.properties</filename> configuration file for jicofo.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    services.prosody = {
+      enable = true;
+      modules = {
+        admin_adhoc = false;
+        bosh = true;
+        ping = true;
+        roster = true;
+        saslauth = true;
+        tls = true;
+      };
+      extraModules = [ "pubsub" ];
+      authentication = "anonymous";
+      extraConfig = ''
+        certificates = "${config.services.prosody.dataDir}"
+      '';
+      virtualHosts.${cfg.hostname} = {
+        enabled = true;
+        domain = cfg.hostname;
+        extraConfig = ''
+          authentication = "anonymous"
+          c2s_require_encryption = false
+          admins = { "focus@auth.${cfg.hostname}" }
+
+          Component "conference.${cfg.hostname}" "muc"
+            storage = "memory"
+
+          Component "jitsi-videobridge.${cfg.hostname}"
+            component_secret = os.getenv("VIDEOBRIDGE_COMPONENT_SECRET")
+        '';
+      };
+      virtualHosts."auth.${cfg.hostname}" = {
+        enabled = true;
+        domain = "auth.${cfg.hostname}";
+        extraConfig = ''
+          authentication = "internal_plain"
+
+          Component "focus.${cfg.hostname}"
+            component_secret = os.getenv("JICOFO_COMPONENT_SECRET")
+        '';
+      };
+    };
+    systemd.services.prosody.serviceConfig.EnvironmentFile = [ "${config.services.prosody.dataDir}/jitsi-meet-secrets" ];
+
+    users.groups.jitsi-meet = {};
+
+    systemd.services.jitsi-meet-init-secrets = {
+      wantedBy = [ "multi-user.target" ];
+      before = [ "prosody.service" "jicofo.service" "jitsi-videobridge.service" ];
+      serviceConfig = {
+        Type = "oneshot";
+      };
+
+      script = let
+        secrets = [ "VIDEOBRIDGE_COMPONENT_SECRET" "JICOFO_COMPONENT_SECRET" "JICOFO_USER_SECRET" ];
+        prosoDir = config.services.prosody.dataDir;
+        prosoPkg = config.services.prosody.package;
+      in
+      ''
+        if [ ! -f /etc/jitsi/jitsi-meet-secrets ]; then
```

Comment:
It can'b be only in `/var/lib/prosody` because it is `drwx------ 3 prosody prosody` so the other components (using `DynamicUser`) wouldn't be able to access it.

Stateful `/etc` - fair point, I'l move this to `/var/lib/jitsi`. Prosody would be able to read it from there too if we add it to the `jitsi-meet` group.